File: PART_3/Composite.java
package PART_3;
import java.util.List;
import java.util.Map;
import java.util.ArrayList;
import java.util.HashMap;
public class Composite {
    public static class Car{
        String engine;
        String tyres;
        public Car(String engine, String tyres){
            this.engine = engine;
            this.tyres = tyres;
        }
        @Override
        public String toString(){
            return "Engine: " + this.engine + "," +
            "Tyres: " + this.tyres;
        }
    }
    public static void main(String[] args) {
        List<String> names = new ArrayList<>();
        names.add("Pablo");
        names.add("Luisa");
        names.add("Castri");
        names.add("Sebas");
        names.add("Daniel");
        names.forEach(name ->{
            System.out.println(name);
        });
        //Class
        Car rb19 = new Car("Honda", "Pirelli P Zero C4");
        System.out.println(rb19);
        // Maps
        Map<String, Float> grades = new HashMap<>();
        grades.put("Pablo", 3.5f);
        grades.put("Luisa", 5f);
        grades.put("Castri", 4.8f);
        grades.put("Sebas", 4.8f);
        grades.put("Dani", 4.81f);
        for(String key : grades.keySet()){
            System.out.println(key + "," + grades.get(key));
        }
    }
}

File: PART_3/Composite.c
#include<stdio.h>
#include<string.h>
//Dict implementation
const int MAX_SIZE = 100;
int size = 0;
char keys[MAX_SIZE][100];
float values[MAX_SIZE];
int getIndex(char key[]){
    for(int i = 0; i < size; i++){
        if(strcmp(keys[i],key) == 0){
            return i;
        }
    }
    return -1;
}

struct Car{
    char engine[15];
    char tyres[20];
};
int main(){
    char names[5][10] = {"Pablo", "Luisa",
        "Castri","Sebas","Daniel"
    };
    for(int i = 0; i < 5; i++){
        printf("%s\n",names[i]);
    }
    struct Car rb19;
    strcpy(rb19.engine, "Honda");
    strcpy(rb19.tyres, "Pirelli P Zero C4");
    printf("Engine: %s\n",rb19.engine);
    printf("Tyres: %s\n",rb19.tyres);
    return 0;
}